### ISSUE TITLE
subsasgn of symbolic array with logical indexing fails when zero or one elements affected

### ISSUE THREAD

**Issue by mtmiller** [cbm755/octsympy#598]
Assignment to a symbolic array using logical indexing fails when the assignment would affect either zero or one elements of the array.

Simple example of assignment to zero elements:

```
>> A = sym ([0 0])
>> indices = [false false]
>> A[indices] = []
error: Python exception: TypeError: cannot add matrix and <type 'int'>
    occurred at line 1 of the Python code block
error: called from
    python_cmd at line 174 column 5
    mat_replace at line 40 column 15
    subsasgn at line 99 column 13
```

Simple example of assignment to one element:

```
>> A = sym ([0 0])
>> indices = [true false];
>> A(indices) = 1;
error: Python exception: AttributeError: 'One' object has no attribute 'T'
    occurred at line 5 of the Python code block
error: called from
    python_cmd at line 174 column 5
    mat_mask_asgn at line 88 column 5
    mat_replace at line 71 column 7
    subsasgn at line 99 column 13
```

If `indices` contains two or more `true` values, the assignment succeeds.

**Comment by genuinelucifer** [comment]
This also occurs if we try to assign `[]` to any number of elements (may be more than 1)...  
```
>> A = sym([0 0 0]);
>> indices = [true false true];
>> A(indices) = [];
error: Python exception: TypeError: cannot add matrix and <type 'int'>
    occurred at line 1 of the Python code block:
    _ins[0].col_del(_ins[1] - 1); return _ins[0],
error: called from
    python_cmd at line 175 column 5
    mat_replace at line 40 column 15
    subsasgn at line 99 column 13
```

**Comment by cbm755** [comment]
This is fixed, thanks @genuinelucifer.

(it didn't auto-close: I think github is rather particular about the string "Fixes #598")
